components/movementsensor/cameramono/cameramono.go

```diff
@@ -118,6 +118,7 @@ type cameramono struct {
 	logger                  golog.Logger
 	result                  result
 	stream                  gostream.VideoStream
+	mu                      sync.RWMutex
```

Comment:
We lock the mutex whenever we read our state, but not when we write it. which means the mutex isn't yet preventing race conditions. Lock it in a few more places, too.

Reply:
Added, let me know if I've missed any.

Reply:
tl/dr: see the very last line.

Part of me says "any time anyone reads or writes any state within the struct (including `result`, `stream`, `motion`, and `lastErr`), they should lock the mutex." but that's overly broad: 
- `stream` is only used in the constructor (no chance of race conditions), the background goroutine, and `Close()`. and if `Close()` causes trouble with the background thread, that's okay because we're closing things and it'll never be used again. So, it's probably fine that this isn't protected by the mutex.
- `motion` is only used by the background goroutine and `extractMovementFromOdometer()`, which is only called from the background goroutine. So, no chance of race conditions here, either. Part of me wonders if it can be made a local variable in `backgroundWorker()` instead of a field in the struct, but that's probably out of scope for the current PR.
- There will only be race conditions around `lastErr` if copying and assigning it are not atomic. I don't know enough about Go's data model to know if that's a good assumption. For peace of mind, I'd protect `lastErr` with the mutex just in case.
- Every piece of `result` except `result.dt` is protected by the mutex. However, `result.dt` appears to be written in one place and never read, so not protecting it seems fine. Maybe it can be removed entirely? but that's probably out of scope for the current PR.

So, I'd lock the mutex whenever reading or writing `lastErr` (and maybe even that is unnecessary), and the rest LGTM!

Reply:
`results.dt` is in the struct in case we ever need to mutate time for the algorithm long term, will lock if ever. And yes other things can be locked/unlocked without a defer but I feel unnecessary.

Will change `lastErr` mutex.